### PR TITLE
Contest

### PR DESCRIPTION
**Pull request by Manjukcthapa**
# Description

Added additional info tot the register endpoint.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] There are no merge conflicts

### CODE REVIEW

File: routes/city.js [hunk 1]
```diff
@@ -309,10 +309,12 @@ router.post("/spec-search", tokenAuthentication, async (req, res) => {
     
     for(var i = 0; i < disID.length; i++) {
       if ( i==0 ) {
-        exitData = filteredSearch.filter(function(city) {
```
Comment (shweps13) [inline routes/city.js]: Why need to delete filter function??
Without filtering we will lose the whole feature.

File: routes/users.js [hunk 2]
```diff
@@ -822,7 +819,7 @@ router.post("/signup", async (req, res) => {
 // Register Handle for web
 router.post("/register", async (req, res) => {
   // check required fields
-  const { name, email, password, location } = req.body;
+  const { name, email, password, location,likes, dislikes,factors} = req.body;
   if (!name || !email || !password || !location) {
```
Comment (shweps13) [inline routes/users.js]: When FE is creating new users, they don't need to send {likes, dislikes, factors} - it's useless code.
{likes, dislikes, factors} will be empty from the beginning and will be filled after another endpoints

File: routes/users.js [hunk 3]
```diff
@@ -855,7 +856,10 @@ router.post("/register", async (req, res) => {
             _id: userSaved._id,
             name: userSaved.name,
             email: userSaved.email,
-            location: userSaved.location
+            location: userSaved.location,
+            likes:userSaved.likes,
+            dislikes:userSaved.dislikes,
+            factors:userSaved.factors
```
Comment (shweps13) [inline routes/users.js]: As I said - we don't need to repeat that a lot of times

File: routes/users.js [hunk 4]
```diff
@@ -845,7 +842,11 @@ router.post("/register", async (req, res) => {
         const newUser = new User({
           name,
           email,
+          likes,
+          dislikes,
+          factors,
           location,
+
```
Comment (shweps13) [inline routes/users.js]: and here

File: routes/users.js (hunk 2 above)
Comment (Manjukcthapa) [reply routes/users.js]: I  had removed those items before from req.body. I don't know how it happens again. Anyway, I had fixed it.

File: routes/users.js (hunk 3 above)
Comment (Manjukcthapa) [reply routes/users.js]: May be while fixing error removed item was updated. Fixed it

File: routes/users.js (hunk 2 above)
Comment (Manjukcthapa) [reply routes/users.js]: I understand what you are trying to say. I have removed items from req.body. while fixing the error it updated. Anyway fixed it

File: routes/users.js (hunk 4 above)
Comment (Manjukcthapa) [reply routes/users.js]: fixed it

File: routes/users.js (hunk 3 above)
Comment (Manjukcthapa) [reply routes/users.js]: fixed it

File: routes/users.js (hunk 4 above)
Comment (Manjukcthapa) [reply routes/users.js]: fixed it

File: routes/city.js (hunk 1 above)
Comment (Manjukcthapa) [reply routes/city.js]: Re-updated

Comment (Manjukcthapa) [reply routes/city.js]: This is not my fault. @IslaMcN  had revome it and merge it into a contest. Anyway, I had fixed it.
https://github.com/Lambda-School-Labs/best-places-to-live-be/pull/153/files

**Pull request merged**